client/my-sites/checkout/index.js

```diff
@@ -32,6 +38,15 @@ export default function () {
 		clientRender
 	);
 	page( '/checkout/jetpack/:productSlug', noSite, checkoutSiteless, makeLayout, clientRender );
+	page(
+		`/checkout/jetpack/:productSlug/${ getLanguageRouteParam() }`,
```

Comment:
You don't need to create a second `/checkout/jetpack/:productSlug` route handler. Just add the lang suffix to the original one. By default it's optional. And `setLocaleMiddleware` will act as noop if the lang param is not present.